native/src/main/java/io/ballerina/stdlib/websocket/WebSocketUtil.java

```diff
@@ -218,6 +222,21 @@ static String getErrorMessage(Throwable err) {
         return err.getMessage();
     }
 
+    public static BString getBString(byte[] byteArray) {
+        return StringUtils.fromString(
+                new String(byteArray, StandardCharsets.UTF_8));
+    }
+
+    public static boolean hasStringType(Type targetType) {
+        List<Type> memberTypes = ((UnionType) targetType).getMemberTypes();
+        return memberTypes.stream().anyMatch(member -> member.getTag() == TypeTags.STRING_TAG);
+    }
+
+    public static boolean hasByteArrayType(Type targetType) {
+        List<Type> memberTypes = ((UnionType) targetType).getMemberTypes();
```

Comment:
wont this accept types like `string[]` too?

Reply:
Fixed with 12d7631